src/lightning/pytorch/trainer/call.py

```diff
@@ -93,9 +94,17 @@ def _call_setup_hook(trainer: "pl.Trainer") -> None:
     trainer.strategy.barrier("post_setup")
 
 
-def _call_configure_sharded_model(trainer: "pl.Trainer") -> None:
-    with trainer.strategy.model_sharded_context():
-        _call_lightning_module_hook(trainer, "configure_sharded_model")
+def _call_configure_model(trainer: "pl.Trainer") -> None:
+    # legacy hook
+    if is_overridden("configure_sharded_model", trainer.lightning_module):
+        with trainer.strategy.model_sharded_context():
+            _call_lightning_module_hook(trainer, "configure_sharded_model")
+
+    # we don't normally check for this before calling the hook. it is done here to avoid instantiating the context
+    # managers
```

Comment:
But the context manager doesn't do anything if the hook doesn't end up being called?

Reply:
Even though what you said is true today, I'd rather avoid this unnecessary step so that we don't risk silent changes if there's a side effect inside FSDP or DeepSpeed in the future. We have tests that the context manager is only called once.